include/matx/transforms/solver_common.h

```diff
@@ -0,0 +1,297 @@
+           // For jobvt: All N rows of V^T are computed
+  REDUCED, // For jobu: The first min(m,n) columns of U are computed
+           // For jobvt: The first min(m,n) rows of V^T are computed
+  NONE     // For jobu: No columns of U are computed
+           // For jobvt: No rows of V^T are computed
+};
+
+namespace detail {
+
+__MATX_INLINE__ char SVDJobToChar(SVDJob job) {
+  switch (job) {
+    case SVDJob::ALL:
+      return 'A';
+    case SVDJob::REDUCED:
+      return 'S';
+    case SVDJob::NONE:
+      return 'N';
+    default:
+      MATX_ASSERT_STR(false, matxInvalidParameter, "Job for SVD not supported");
+      return '\0';
+  }
+}
+
+/* Solver utility functions */
+
+enum class BatchType {
+  Vector = 1,
+  Matrix = 2
+};
+
+/**
+ * @brief Sets batch pointers for a batched tensor of arbitrary rank.
+ * 
+ * Clears the given batch pointers vector and then populates it
+ * with pointers to the data of the tensor for batched operations.
+ * Handles both batched matrices and vectors.
+ * 
+ * @tparam BTYPE
+ *   Whether the input is a batch of matrices or vectors
+ * @tparam TensorType
+ *   Type of input tensor a
+ * @tparam PointerType
+ *   Tensor value type
+ * 
+ * @param a
+ *   The tensor for which batch pointers are to be set.
+ * @param batch_ptrs
+ *   The vector to be filled with pointers
+ */
+template <BatchType BTYPE, typename TensorType, typename PointerType = typename TensorType::value_type>
+__MATX_INLINE__ void SetBatchPointers(const TensorType &a, std::vector<PointerType *> &batch_ptrs)
+{
+  MATX_NVTX_START("", matx::MATX_NVTX_LOG_INTERNAL)
+
+  batch_ptrs.clear();
+  
+  if constexpr (BTYPE == BatchType::Vector && TensorType::Rank() == 1) {
+    // single vector case
+    batch_ptrs.push_back(&a(0));
+  }
+  else if constexpr (BTYPE == BatchType::Matrix && TensorType::Rank() == 2) {
+    // single matrix case
+    batch_ptrs.push_back(&a(0, 0));
+  }
+  else {
+    // batched vectors or matrices
+    using shape_type = typename TensorType::desc_type::shape_type;
+    int batch_offset = static_cast<int>(BTYPE);
+    cuda::std::array<shape_type, TensorType::Rank()> idx{0};
+    auto a_shape = a.Shape();
+    size_t total_iter = std::accumulate(a_shape.begin(), a_shape.begin() + TensorType::Rank() - batch_offset, 1, std::multiplies<shape_type>());
+    for (size_t iter = 0; iter < total_iter; iter++) {
+      auto ap = cuda::std::apply([&a](auto... param) { return a.GetPointer(param...); }, idx);
+      batch_ptrs.push_back(ap);
+      UpdateIndices<TensorType, shape_type, TensorType::Rank()>(a, idx, batch_offset);
+    }
+  }
+}
+
+/**
+ * Get a transposed view of a tensor into a user-supplied buffer
+ *
+ * @param tp
+ *   Pointer to pre-allocated memory
+ * @param a
+ *   Tensor to transpose
+ * @param stream
+ *   CUDA stream
+ */
+template <typename TensorType, typename Executor>
+__MATX_INLINE__ auto
+TransposeCopy(typename TensorType::value_type *tp, const TensorType &a, const Executor &exec)
```

Comment:
This function seems like it should go into tensor_utils.h maybe

Reply:
Moved it there